src/scheduler/stat.rs

```diff
@@ -1,13 +1,24 @@
+//! Statistics for work packets
+use super::work_counter::{WorkCounter, WorkCounterBase, WorkDuration};
 use std::any::TypeId;
 use std::collections::HashMap;
 use std::sync::atomic::{AtomicBool, Ordering};
-use std::time::{Duration, SystemTime};
 
+/// Merge and print the work-packet level statistics from all worker threads
 #[derive(Default)]
 pub struct SchedulerStat {
+    /// Map work packet type IDs to work packet names
     work_id_name_map: HashMap<TypeId, &'static str>,
+    /// Count the number of work packets executed for different types
     work_counts: HashMap<TypeId, usize>,
-    work_durations: HashMap<TypeId, Vec<Duration>>,
+    /// Collect work counters from work threads.
+    /// Two dimensional vectors are used.
+    /// The first dimension is for different types of work counters.
+    /// The second dimension if for work counters of the same type but from
+    /// different threads.
+    /// We assume different threads have the same set of work counters
+    /// (in the same order).
+    work_counters: HashMap<TypeId, Vec<Vec<Box<dyn WorkCounter>>>>,
```

Comment:
Even with comments, I find this hard to understand. 
* For the values in the hashmap, `value[0]` is counters for a certain type, and `value[0][0]` is a counter for a certain type for a certain thread. Is it correct?
* Is `value[i][0]` always from the same thread? How does the thread get mapped to the array index?

Reply:
1. Yes.

2. The order of insertion  is determined by when `merge` is called for each local stat.

Reply:
I will try to make it clearer